plugin/trino-raptor-legacy/src/main/java/io/trino/plugin/raptor/legacy/metadata/ShardPredicate.java

```diff
@@ -144,15 +129,15 @@ public static ShardPredicate create(TupleDomain<RaptorColumnHandle> tupleDomain)
                 }
 
                 StringJoiner rangePredicate = new StringJoiner(" AND ", "(", ")").setEmptyValue("true");
-                if (minValue != null) {
+                if (!range.isLowUnbounded()) {
```

Comment:
nit: consider adding `isLow/HighBounded`. May be more readable than `!`

Reply:
That's a good though. I had same feeling when working on this.
i don't want to have two kinds iof methods (`is{Low,High}Unbounded` and `is{Low,High}Bounded`), but i can imaging we could prefer one over the other.

i added `isXxxUnbounded` version because
- `io.trino.spi.predicate.Marker#isLowerUnbounded`, `io.trino.spi.predicate.Marker#isUpperUnbounded` (i consciously moved from lower/upper to low/high, but kept "Unbounded")
- `io.trino.spi.type.VarcharType#isUnbounded`

OTOH i see that most callers have the logic `! isXxxUnbounded`, which ends up being double negation.

Thoughts?

Reply:
More thinking about it I think having `unbounded` seems more natural than `bounded`. As `un` prefix is not about negation, but rather directly saying that value is +/- infinity.
Let's leave it. No idea how to make it look nicer. The problem is with java syntax and using `!` for boolean negation.